corehq/apps/userreports/expressions/list_specs.py

```diff
@@ -122,11 +122,21 @@ def __call__(self, doc, context=None):
         items = _evaluate_items_expression(self._items_expression, doc, context)
 
         try:
-            return sorted(
-                items,
-                key=lambda i: self._sort_expression(i, context),
-                reverse=True if self.order == self.DESC else False
+            items_and_keys = [(item, self._sort_expression(item, context)) for item in items]
+            items_with_empty_sort_keys = [
+                item_and_key[0] for item_and_key in items_and_keys if item_and_key[1] is None
+            ]
+            items_and_nonempty_keys = [
+                item_and_key for item_and_key in items_and_keys if item_and_key[1] is not None
+            ]
+            sorted_items_and_nonempty_keys = sorted(
+                items_and_nonempty_keys,
+                key=lambda item_and_key: item_and_key[1],
+                reverse=True if self.order == self.DESC else False,
             )
+            return items_with_empty_sort_keys + [
+                item_and_key[0] for item_and_key in sorted_items_and_nonempty_keys
+            ]
```

Comment:
Is the change in behavior intentional? Now it always sorts nulls first where previously it sorted them last if `self.order == self.DESC`.

What do you think of this instead
```py
def sort_key(item):
    value = self._sort_expression(item, context)
    # swap 0 and 1 to sort nulls last instead of first
    return (0 if value is None else 1), value
    
return sorted(
    items,
    key=sort_key,
    reverse=True if self.order == self.DESC else False
)
```

Reply:
Your suggestion is way better - updated: 599a86f1a9839b4f80beaa99cae7ee4a9bb7d7b8